examples/plasticity/WellboreProblem_PoroDruckerPrager.xml

```diff
@@ -0,0 +1,286 @@
+<!--
+Problem: wellbore problem subjected to mixed mode loading and anisotropic far-field stresses
+Created by: Jian Huang
+Last update: July 23 2021
+Objective: to verify the poroelastic solver in GEOSX against analytical solution
+Reference: Cheng, A. H. D. (2016). Poroelasticity (Vol. 877). Switzerland: Springer International Publishing.
+-->
+
+
+<Problem>
+<!-- SPHINX_WELLBORE_TABLE -->
+<Functions>
+  <TableFunction 
+    name="timeFunction"
+    inputVarNames="{time}"
+    coordinates="{0.0, 0.1, 1e6}"
+    values="{0.0, 1.0, 1.0}" 
+  />
+</Functions>
+<!-- SPHINX_WELLBORE_TABLE_END -->
+
+<!-- SPHINX_WELLBORE_MESH -->
+<Mesh>
+  <InternalWellbore
+    name="mesh1"
+    elementTypes="{ C3D8 }"
+    radius="{ 0.1, 5.0 }"
+    theta="{ 0, 180 }"
+    zCoords="{ -1, 1 }"
+    nr="{ 40 }"
+    nt="{ 80 }"
+    nz="{ 2 }"
+    trajectory="{ { 0.0, 0.0, -1.0 },
+                  { 0.0, 0.0,  1.0 } }"
+    autoSpaceRadialElems="{ 1 }"
+    useCartesianOuterBoundary="0"
+    cellBlockNames="{ cb1 }"/>
+</Mesh> 
+<!-- SPHINX_WELLBORE_MESH_END -->
+
+<!-- SPHINX_WELLBORE_MATERIAL -->
+<Constitutive>
+  <PorousExtendedDruckerPrager
+    name="porousRock"
+    solidModelName="rock"
+    porosityModelName="rockPorosity"
+    permeabilityModelName="rockPerm"
+  />
+  <ExtendedDruckerPrager 
+    name="rock"
+    defaultDensity="2700"
+    defaultBulkModulus="11111111111.1"
+    defaultShearModulus="8333333333.33"
+    defaultCohesion="1.0e6"
+    defaultInitialFrictionAngle="15.27"
+    defaultResidualFrictionAngle="23.05"
+    defaultDilationRatio="1.0"
+    defaultHardening="0.01"
+  />
+  <CompressibleSinglePhaseFluid 
+    name="water"
+    defaultDensity="1000"
+    defaultViscosity="0.001"
+    referencePressure="0e6"
+    referenceDensity="1000"
+    compressibility="2.09028227021e-10"
+    referenceViscosity="0.001"
+    viscosibility="0.0"
+  />
+  <BiotPorosity
+    name="rockPorosity"
+    grainBulkModulus="1.0e27"
+    defaultReferencePorosity="0.3"
+  />    
+  <ConstantPermeability
+    name="rockPerm"
+    permeabilityComponents="{1.0e-20, 1.0e-20, 1.0e-20}"
+  /> 
+</Constitutive>
+<!-- SPHINX_WELLBORE_MATERIAL_END -->
+
+<!-- SPHINX_WELLBORE_ELEMENTREGIONS -->
+<ElementRegions>
+  <CellElementRegion 
+    name="Omega" 
+    cellBlocks="{cb1}" 
+    materialList="{ rock, water, porousRock, rockPerm, rockPorosity}"
```

Comment:
@CusiniM, I could see the need to specify all of the sub-materials independently becoming a usability issue.  Users may perhaps forget to add rock when they define porousRock.  Just something to think about.

Reply:
I don't love this either. It would be nice if specifying only the higher constitutive model in the hierarchy (porousRock) was enough. It is not the way the code was initially designed though so this may need some rethinking.